tty/imx6ull-uart/imx6ull-uart.c

```diff
@@ -361,41 +363,47 @@ static void uart_process_rx(void)
 }
 
 
-static void uart_intrthr(void *arg)
+static void uart_process_tx(void)
 {
-	uint8_t flags;
+	int wake = 0;
+
+	while (libtty_txready(&uart.tty_common)) {
+		if ((*(uart.base + uts) & UTS_TXFULL)) {
+			*(uart.base + ucr1) |= UCR1_TRDYEN;
+			break;
+		}
+
+		/* FIXME: potential data race on tx_fifo (lock-free access) */
+		*(uart.base + utxd) = libtty_popchar(&uart.tty_common);
```

Comment:
Not totally without a lock - uart.lock is taken in uart_intrthr

Reply:
This comment refers to `fifo_pop_back` on TX FIFO which is not locked by libtty (tty->tx_mutex) and can lead to potential data race with `fifo_push` on TX FIFO.